config/initializers/lazy_load_routes.rb

```diff
@@ -0,0 +1,42 @@
+# To lazy load routes:
+# 1) Modify the RoutesReloader to skip normal reload! unless opted-in or Rails server
+# 2) Insert a Rack::Middleware to catch the first request with empty routes, opt-in and reload!
+# 3) Modify the RoutesInspector used by rake routes to detect empty routes, opt-in and reload!
+module RoutesReloaderLazyLoadRoutes
```

Comment:
Super minor, but can we structure these like to keep them associated?

```ruby
module LazyLoadedRoutes
  module RoutesReloader; end
  module RoutesMiddleware; end
  module RoutesInspector; end
end
  
```

Reply:
yeah, I like that... I'm hoping cross repo tests get "close" to passing this time around...

🤷‍♂ [Nope](https://travis-ci.org/ManageIQ/manageiq-cross_repo-tests/jobs/653590343?utm_medium=notification&utm_source=github_status)

Reply:
so, the problem with my current approach is the URL helper "magic" for routes aren't defined if the routes aren't yet defined, leading to UI/api test failures like this:

```
  1) Pictures As an attribute Queries of Service Templates allows queries of the related picture and image_href
     Failure/Error: get api_service_template_url(nil, template), :params => { :attributes => "picture,picture.image_href" }

     NoMethodError:
       undefined method `api_service_template_url' for #<RSpec::ExampleGroups::Pictures::AsAnAttribute::QueriesOfServiceTemplates:0x00007fefd9b48fb8>
     # /Users/joerafaniello/.gem/ruby/2.6.5/gems/actionpack-5.1.7/lib/action_dispatch/testing/assertions/routing.rb:175:in `method_missing'
     # /Users/joerafaniello/.gem/ruby/2.6.5/gems/actionpack-5.1.7/lib/action_dispatch/testing/integration.rb:398:in `method_missing'
     # ./spec/requests/pictures_spec.rb:59:in `block (4 levels) in <top (required)>'
```

Reply:
So we can probably force loading the routes in RSpec when we configure it, probably via a before(:suite) or something.  Somewhere in here - https://github.com/ManageIQ/manageiq-ui-classic/blob/master/spec/spec_helper.rb#L48-L71